plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,256 @@
+package plugins
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/openlyinc/pointy"
+	"github.com/stellar/kelp/queries"
+	"github.com/stellar/kelp/support/utils"
+
+	"github.com/stellar/go/txnbuild"
+
+	hProtocol "github.com/stellar/go/protocols/horizon"
+	"github.com/stellar/kelp/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func makeTestVolumeFilterConfig(baseCapInBase, baseCapInQuote float64, additionalMarketIDs, optionalAccountIDs []string, mode volumeFilterMode) *VolumeFilterConfig {
+	var baseCapInBasePtr *float64
+	if baseCapInBase >= 0 {
+		baseCapInBasePtr = pointy.Float64(baseCapInBase)
+	}
+
+	var baseCapInQuotePtr *float64
+	if baseCapInQuote >= 0 {
+		baseCapInQuotePtr = pointy.Float64(baseCapInQuote)
+	}
+
+	return &VolumeFilterConfig{
+		SellBaseAssetCapInBaseUnits:  baseCapInBasePtr,
+		SellBaseAssetCapInQuoteUnits: baseCapInQuotePtr,
+		mode:                         mode,
+		additionalMarketIDs:          additionalMarketIDs,
+		optionalAccountIDs:           optionalAccountIDs,
+	}
+}
+
+func makeTestVolumeFilter(config *VolumeFilterConfig, additionalMarketIDs, optionalAccountIDs []string) *volumeFilter {
+	marketID := MakeMarketID("", "native", "native")
+	marketIDs := utils.Dedupe(append([]string{marketID}, additionalMarketIDs...))
+
+	query, e := queries.MakeDailyVolumeByDateForMarketIdsAction(&sql.DB{}, marketIDs, "sell", optionalAccountIDs)
+	if e != nil {
+		panic(e)
+	}
+
+	return &volumeFilter{
+		name:                   "volumeFilter",
+		baseAsset:              utils.NativeAsset,
+		quoteAsset:             utils.NativeAsset,
+		config:                 config,
+		dailyVolumeByDateQuery: query,
+	}
+}
+
+func TestMakeFilterVolume(t *testing.T) {
+	testAssetDisplayFn := model.MakeSdexMappedAssetDisplayFn(map[model.Asset]hProtocol.Asset{model.Asset("XLM"): utils.NativeAsset})
+
+	testCases := []struct {
+		name       string
+		marketIDs  []string
+		accountIDs []string
```

Comment:
test cases should take a `want` variable. the inputs should be structured so the want values are different

Reply:
I don't think this is possible. the `want` variable requires an input configuration. Our current approach, which loops over modes and configs, forecloses referencing it in the `testCases` array - the config doesn't exist at this point.

Within our current test structure, we'd have to define the testCases within the inner loop of our array. We could do this, but it's a bit messy. Let me know.